.github/workflows/build-and-deploy.yml

```diff
@@ -16,6 +16,10 @@ jobs:
       - name: Checkout
         uses: DanySK/action-checkout@0.2.1
       - uses: DanySK/build-check-deploy-gradle-action@2.1.5
+        env:
+          GH_TOKEN: ${{ secrets.GH_TOKEN }}
+          BB_USER: ${{ secrets.BB_USER }}
+          BB_TOKEN: ${{ secrets.BB_TOKEN }}
```

Comment:
I'm not sure if you can pass environment variables like this. Actually, I think you can not...

Reply:
I tried on my fork and it works. Maybe it's not the best way... what do you suggest?

Reply:
If it works, then it is actually fine

Reply:
It's strange: the variables are correctly set otherwise it would throw:
```
NullPointerException: Error while retrieving environment variables: variables not defined in the system environment.
```

instead, a 401 Unauthorized response is returned from GitHub and Bitbucket, like the contents of the variables are not correct. It seems they are not correctly interpolated, but I can't figure out why: according to [this](https://docs.github.com/en/actions/learn-github-actions/environment-variables#about-environment-variables) page is possible to pass environment variables in a step and I don't think any of the limitations described there could cause the error.

Reply:
I'm unsure what GH Actions passes through, but for sure they can't be my secrets; that would expose my repo to a PR attack